Classes/common/query/CDTQIndexUpdater.m

```diff
@@ -397,12 +396,14 @@ + (CDTQSqlParts *)createPartsForFieldNames:(NSArray *)fieldNames
             [CDTQValueExtractor extractValueForFieldName:fieldName fromDictionary:rev.body];
 
         if (value) {
+            // Do not include an empty array field as you would not include a field with no value.
+            if ([value isKindOfClass:[NSArray class]] && ((NSArray *)value).count == 0) {
+                continue;
+            }
+            
             [includedFieldNames addObject:fieldName];
             [args addObject:value];
```

Comment:
Same question from sync-android PR. Doesn't value get added to the args array twice?  And does that matter?